kymatio/scattering1d/frontend/base_frontend.py

```diff
@@ -581,6 +583,107 @@ def create_filters(self):
+                path['coef'], self.ind_start[res], self.ind_end[res])
+        path['coef'] = self.backend.reshape_output(
+            path['coef'], batch_shape, n_kept_dims=1)
+        S = [{**path, 'order': 0}]
+
+        # First and second order
+        for path in S_gen:
+            # "The phase of the integrand must be set to a constant. This
+            # freedom in setting the stationary phase to an arbitrary constant
+            # value suggests the existence of a gauge boson" — Glinsky
+            path['coef'] = self.backend.modulus(path['coef'])
+
+            # Temporal averaging and unpadding. Switch cases:
+            # 1. If averaging is global, no need for unpadding at all.
+            # 2. If averaging is local, averaging depends on order:
+            #     2a. at order 1, unpad at resolution log2_T
+            #     2b. at order 2, average and unpad at resolution log2_T
+            # 3. If there is no averaging, unpadding depends on order:
+            #     3a. at order 1, unpad at resolution log2_T
+            #     3b. at order 2, unpad at resolution j2
+            # (for simplicity, we assume oversampling=0 in the rationale above,
+            #  but the implementation below works for any value of oversampling)
+            if self.average == 'global':
+                # Case 1.
+                path['coef'] = self.backend.average_global(path['coef'])
+            else:
+                if len(path['n']) > 1 and not self.average:
+                    # Case 3b.
+                    res = max(path['j'][-1] - self.oversampling, 0)
+                else:
+                    # Cases 2a, 2b, and 3a.
+                    res = max(self.log2_T - self.oversampling, 0)
+                    if len(path['n']) > 1 and self.average == 'local':
+                        # Case 2b.
+                        path = time_averaging(
+                            path, self.backend, self.phi_f, self.oversampling)
+                # Cases 2a, 2b, 3a, and 3b.
+                path['coef'] = self.backend.unpad(
+                    path['coef'], self.ind_start[res], self.ind_end[res])
+
+            # Frequential averaging and unpadding
+            if self.average_fr and not (path['spin'] == 0):
+                path = frequency_averaging(path, self.backend,
+                    self.filters_fr[0], self.oversampling_fr, self.average_fr)
+            if not (self.out_type == 'array' and self.format == 'joint'):
+                # TODO unpad_frequency
+                # path['coef'] = backend.unpad_frequency(path['coef'],
+                #    path['n1_max'], path['n1_stride'])
+                raise NotImplementedError
+
+            # Splitting and reshaping
+            if self.format == 'joint':
+                paths = [path]
+            elif self.format == 'time':
+                # TODO split
+                # paths = self.backend.split(path)
+                raise NotImplementedError
+            for path in paths:
```

Comment:
why make the forloop if there is only one element in list "paths"? different cases in self.format == "time"?

Reply:
That's right. Under `self.format == 'time'`, we will apply `self.backend.split(path)` so as to split the 3D array `(batch, freq, time)` into a list of 2D arrays `(batch, time)`. This list will have more than one element.

```python
if self.format == 'joint':
    paths = [path]
elif self.format == 'time':
    paths = self.backend.split(path)
for path in paths:
    path['coef'] = self.backend.reshape_output(
        path['coef'], batch_shape, n_kept_dims=2)
    S.append({**path, 'order': len(path['n'])})
```

So i came up with this `paths = [path]` solution under `self.format == 'joint'` (current PR). This way, the subsequent lines (`reshape_output` and `S.append`) are agnostic to `self.format`.

The other solution would have been to duplicate the code:

```python
if self.format == 'joint':
    path['coef'] = self.backend.reshape_output(
        path['coef'], batch_shape, n_kept_dims=2)
    S.append({**path, 'order': len(path['n'])})
elif self.format == 'time':
    paths = self.backend.split(path)
    for path in paths:
        path['coef'] = self.backend.reshape_output(
            path['coef'], batch_shape, n_kept_dims=2)
        S.append({**path, 'order': len(path['n'])})
```

Reply:
ok, that's very reasonable.